shortcode.php

```diff
@@ -330,3 +330,110 @@ function geoip_detect_shortcode_user_info($attr) {
+ * `[geoip_detect2_hide_unless country="US" region="Texas"]<h1>Title</h1>[/geoip_detect2_hide_unless]`
+ * Only displays the inner content if the specified attributes match the visitor's country and region/state.
+ * 
+ * `[geoip_detect2_hide_unless country="US" state="TX" and-not="Houston"]<h1>Title</h1>[/geoip_detect2_hide_unless]`
+ * Only displays the inner content if the specified attributes match the visitor's country and region/state,
+ * and the specified exclusion is not found in any possible GeoIP Detect field (e.g. visitor's country,
+ * city, timezone, etc.).
+ * 
+ * `[geoip_detect2_hide_unless contenant="North America"]<h1>Title</h1>[/geoip_detect2_hide_unless]`
+ * Only displays the inner content if the specified attributes match the visitor's contenant.
+ * 
+ */
+function geoip_detect2_shortcode_hide_unless($atts, $content = null) {
+    $atts_array = shortcode_atts(array(
+        'timezone' => null,
+        'continent' => null,
+        'country' => null,
+        'region' => null,
+        'state' => null,
+        'city' => null,
+        'and-not' => null),
+        $atts);
+
+    $info = geoip_detect2_get_info_from_current_ip();
+    $criteria_test_flag = true;       // If set to false, nothing will display
+    
+    /* Attribute Conditions */
+    // Timezone
+    if ($atts_array['timezone'] != null) {
+        if ($info->location->timeZone && $atts_array['timezone'] != $info->location->timeZone) {
+            $criteria_test_flag = false;
+        }
+    }
+
+    // Continent
+    if ($atts_array['continent'] != null) {
+        if ($info->continent->code && $atts_array['continent'] != $info->continent->code) {
+            $criteria_test_flag = false;
+        }
+    }
+
+    // Country
+    if ($atts_array['country'] != null) {
+        if ($info->country->name && $atts_array['country'] != $info->country->name
+            && $info->country->isoCode && $atts_array['country'] != $info->country->isoCode) {
+            $criteria_test_flag = false;
+        }
+    }
+
+    // Region/State
+    if ($atts_array['region'] != null) {
+        if ($info->mostSpecificSubdivision->name && $atts_array['region'] != $info->mostSpecificSubdivision->name
+        && $info->mostSpecificSubdivision->isoCode && $atts_array['region'] != $info->mostSpecificSubdivision->isoCode) {
+            $criteria_test_flag = false;
+        }
+    }
+    if ($atts_array['state'] != null) {
+        if ($info->mostSpecificSubdivision->name && $atts_array['state'] != $info->mostSpecificSubdivision->name
+            && $info->mostSpecificSubdivision->isoCode && $atts_array['state'] != $info->mostSpecificSubdivision->isoCode) {
+            $criteria_test_flag = false;
+        }
+    }
+
+    // City
+    if ($atts_array['city'] != null) {
+        if ($info->city->name && $atts_array['city'] != $info->city->name) {
+            $criteria_test_flag = false;
+        }
+    }
+
+    // And Not (Exclusion)
+    if ($atts_array['and-not'] != null) {
```

Comment:
I'm not convinced of such an unspecific way of specifying exclusion criteria. What if a continent ISO code is at the same time a country or region Iso code or so?

Would it be possible to solve such case by nesting shortcodes?